src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_io_net_stm32f4_nucleo_wifi.c

```diff
@@ -107,28 +108,38 @@ xi_bsp_io_net_state_t xi_bsp_io_net_read( xi_bsp_socket_t xi_socket,
 
     *out_read_count = 0;
 
-    if ( xi_wifi_state == xi_wifi_state_connected )
+    if ( xi_net_state.wifi_state == xi_wifi_state_connected )
     {
-        if ( xi_wifi_buffer.size == 0 )
+        if ( xi_net_state.head == NULL )
         {
             return XI_BSP_IO_NET_STATE_BUSY;
         }
-        else if ( xi_wifi_buffer.size > count )
-        {
-            return XI_BSP_IO_NET_STATE_ERROR;
-        }
         else
         {
-            memcpy( buf, xi_wifi_buffer.bytes, xi_wifi_buffer.size );
-            *out_read_count     = xi_wifi_buffer.size;
-            xi_wifi_buffer.size = 0;
+            xi_data_desc_t* head = xi_net_state.head;
+
+            size_t bytes_available = head->length - head->curr_pos;
+            size_t bytes_to_copy = bytes_available > count ? count : bytes_available;
```

Comment:
or even you could use a macro for selecting the minimum of two integrals:
#define MIN( a, b ) ( ( ( a ) < ( b ) ) ? ( a ) : ( b ) )